api/v1alpha1/openapi.yaml

```diff
@@ -1307,6 +1307,390 @@ paths:
+      description: delete a Certificate Signing Request
+      operationId: deleteCertificateSigningRequest
+      parameters:
+      # TODO - k8s also has dryRun, gracePeriodSeconds, orphanDependents, propagationPolicy
+      # https://github.com/kubernetes/kubernetes/blob/f8d5b2074c9b3a1629ef085b9e17a5370f393732/api/openapi-spec/v3/apis__certificates.k8s.io__v1_openapi.json#L1862
+        - name: name
+          in: path
+          description: name of the Certificate Signing Request
+          required: true
+          schema:
+            type: string
+      responses:
+        "200":
+          description: OK
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/CertificateSigningRequest'
+        "202":
+          description: Accepted
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/CertificateSigningRequest'
+        "401":
+          description: Unauthorized
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+  /api/v1/certificatesigningrequests/{name}/status:
+    # TODO - k8s also has a patch method
+    # https://github.com/kubernetes/kubernetes/blob/f8d5b2074c9b3a1629ef085b9e17a5370f393732/api/openapi-spec/v3/apis__certificates.k8s.io__v1_openapi.json#L2581
+    get:
+      tags:
+        - certificatesigningrequest
+      description: read status of the specified CertificateSigningRequest
+      operationId: readCertificateSigningRequestStatus
+      parameters:
+      # TODO - in k8s there is also a pretty printing parameter
+      # https://github.com/kubernetes/kubernetes/blob/f8d5b2074c9b3a1629ef085b9e17a5370f393732/api/openapi-spec/v3/apis__certificates.k8s.io__v1_openapi.json#L2572
+        - name: name
+          in: path
+          description: name of the CertificateSigningRequest
+          required: true
+          schema:
+            type: string
+            uniqueItems: true # differs from enrollmentRequest
+      responses:
+        "200":
+          description: OK
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/CertificateSigningRequest'
+        "401":
+          description: Unauthorized
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+    put:
+      tags:
+        - certificatesigningrequest
+      description: replace status of the specified CertificateSigningRequest
+      operationId: replaceCertificateSigningRequestStatus
+      parameters:
+      # TODO - k8s also has dryRun, fieldValidation, fieldManager
+      # https://github.com/kubernetes/kubernetes/blob/f8d5b2074c9b3a1629ef085b9e17a5370f393732/api/openapi-spec/v3/apis__certificates.k8s.io__v1_openapi.json#L2705
+        - name: name
+          # TODO - k8s does not have this parameter - keep?
```

Comment:
Yes. That's the `{name}` in  `/api/v1/certificatesigningrequests/{name}/status`.

Reply:
That's what I thought - I don't understand how k8s is missing this for PUT under `/{name}/status`. They do have it for GET.